lib/rucio/transfertool/cms_fts3.py

```diff
@@ -0,0 +1,358 @@
+# Copyright European Organization for Nuclear Research (CERN)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Authors:
+# - Diego Ciangottini, <ciangottini@pg.infn.it>, 2018
+
+from __future__ import absolute_import
+import logging
+import os
+import sys
+import subprocess
+import tempfile
+from datetime import timedelta
+from hashlib import sha1
+from socket import gaierror
+
+import requests
+from requests.packages.urllib3 import disable_warnings  # pylint: disable=import-error
+from requests.exceptions import Timeout, RequestException, ConnectionError, SSLError, HTTPError
+
+from dogpile.cache import make_region
+from dogpile.cache.api import NoValue
+
+from fts3.rest.client.exceptions import BadEndpoint, ClientError, ServerError
+import fts3.rest.client.easy as fts
+from myproxy.client import MyProxyClient, MyProxyClientGetError, MyProxyClientRetrieveError
+
+from rucio.common.config import config_get, config_get_bool
+from rucio.core.monitor import record_counter
+from rucio.transfertool.fts3 import FTS3Transfertool
+
+
+logging.getLogger("requests").setLevel(logging.CRITICAL)
+disable_warnings()
+
+logging.basicConfig(stream=sys.stdout,
+                    level=getattr(logging,
+                                  config_get('common', 'loglevel',
+                                             raise_exception=False,
+                                             default='DEBUG').upper()),
+                    format='%(asctime)s\t%(process)d\t%(levelname)s\t%(message)s')
+
+REGION_SHORT = make_region().configure('dogpile.cache.memory',
+                                       expiration_time=1800)
+
+
+class CMSUserTransfer(FTS3Transfertool):
```

Comment:
In principle this is a generic transfertool that happens to deal with MyProxy. Maybe better to call it something like `FTS3MyProxyTransfertool` or something similar? Essentially making CMS the first "client" of this transfertool.

Reply:
well, the key definition for retrieving the proxy from myproxy is quite CRAB bound, nevertheless I think it can be overwritten in other subclasses if needed. So agreed.